include/slam_toolbox/serialization.hpp

```diff
@@ -34,11 +34,12 @@ inline bool fileExists(const std::string & name)
   return stat(name.c_str(), &buffer) == 0;
 }
 
+template<class NodeT>
 inline bool write(
   const std::string & filename,
   karto::Mapper & mapper,
   karto::Dataset & dataset,
-  rclcpp::Node::SharedPtr node)
+  NodeT && node)
```

Comment:
Why not keep it as a shared pointer?

Reply:
MergeMapsKinematic (as Node) is using serialization::read, modified both `read` and `write` use with both Node and LifecycleNode.

<https://github.com/SteveMacenski/slam_toolbox/blob/78af8485ebfd7d2b95e50188a341b86b4069e644/src/merge_maps_kinematic.cpp#L75>